### DISCUSSION TITLE
Is it possible to read/write atom outside React component?

### DISCUSSION THREAD

**Discussion by dvakatsiienko** [pmndrs/zustand#1367] [Q&A]
I use a library that exposes the data using non-react-component callbacks.

I need to save a ref to HTML element inside of that callback.
But the problem is that it is not a React Component.

It is possible to write to Jotai atom from a regular function?

Example:
```typescript
const useStore = create((set) => ({
  elementRef: null,
  setElementRef: (nextElementRef) => set((state) => ({ elementRef: nextElementRef })),
}));

beforeEventEdit(event) { // ← NOT a React Component
  // event.elementRef — holds a ref to a target HTML element

   const setElementRef = useStore((state) => state.setElementRef)

  setElementRef(event.elementRef);
},
```

**Comment by dai-shi** [comment]
> It is possible to write to Jotai atom from a regular function?

Do you mean zustand store?
```js
  const { setElementRef } = useStore.getState();
```
